api/src/services/report/smsparser.js

```diff
@@ -237,6 +254,35 @@ exports.parse = (def, doc) => {
     }
   }
 
+  if(Object.keys(def.fields).some(key => def.fields[key].type === 'bsYear')) {
+    let bsYear;
+    let bsMonth;
+    let bsDay;
+    for (const k of Object.keys(def.fields)) {
+      switch (def.fields[k].type) {
+      case 'bsYear':
+        bsYear = msgData[k];
+        break;
+      case 'bsMonth':
+        bsMonth = msgData[k];
+        break;
+      case 'bsDay':
+        bsDay = msgData[k];
+        break;
+      }
+    }
+
+    if (bsYear && bsMonth && bsDay) {
+      try {
+        formData.lmpDate = moment(bs.toGreg_text(bsYear, bsMonth, bsDay)).valueOf();
```

Comment:
I've changed the field to `lmp_date`.

One side effect of using it is that when using the new `bsDate` type (i.e. B form in the description), the report in the app shows the `bsDate` field as milliseconds since epoch instead of the date format.

It is fine when the type is `date` (form A and C in the description).
![image](https://user-images.githubusercontent.com/20041732/150578481-980cf378-1d26-4b47-b7e9-6ce77a1b87ba.png)

If I change the form type from `B` to `A` in the document, it is then displayed correctly.
Do we need to specify somewhere that `bsDate` is a date field so that it is formatted properly?

Reply:
Added `bsDate` to the date formatter:
https://github.com/medic/cht-core/blob/29cc1e93bf37569fc6b0e0d5a16c891c2b8ebc51/webapp/src/ts/services/format-data-record.service.ts#L259
Now it is formatted properly.